functions-and-scope.js

```diff
@@ -1,83 +1,223 @@
+console.log("averageGrade([6, 4, 5]): " + averageGrade([6, 4, 5]));
+console.log("averageGrade([8, 9, 4, 6, 10]): " + averageGrade([8, 9, 4, 6, 10]));
+
 // ---- Verwachte uitkomsten:
 // averageGrade(grades) geeft 6.642857142857143
 // averageGrade([6, 4, 5]) geeft xxxx
 // averageGrade([8, 9, 4, 6, 10]) geeft xxxx
 
 
 /* 2c: Afronden op twee decimalen */
-// Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
+// Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie
+// netjes wordt afgerond op twee decimalen.
 // Tip: Google is your best friend!
 
-
-
+function roundToTwoDecimal(number) {
+    let numberOfDecimalPlaces = number.toString().length - (number.toString().indexOf(".") + 1);
+    let numberToString = number.toString();
+    let numberSeparated = numberToString.split(".");
+    let numberBeforeTheDecimalPoint = numberSeparated[0];
+    let theNumberAfterTheDecimalPoint = numberSeparated[1];
+    let numberOfPlacesBeforeTheDecimalPoint = numberBeforeTheDecimalPoint.length;
+    // theNumberAfterTheDecimalPoint.split("\t");
+    let arrayAfterTheDecimalPoint = [];
+    let arrayBeforeTheDecimalPoint = [];
+
+   for (let i = 0; i < numberOfDecimalPlaces; i++) {
+        arrayAfterTheDecimalPoint[i] = Number(theNumberAfterTheDecimalPoint[i]);
+   }
+
+   for (let i = 0; i < numberOfPlacesBeforeTheDecimalPoint; i++){
+       arrayBeforeTheDecimalPoint[i] = Number(numberBeforeTheDecimalPoint[i]);
+   }
+
+   for (let i = numberOfDecimalPlaces - 1; i > 1; i-- ){
+       if (arrayAfterTheDecimalPoint[i] > 4){
+           arrayAfterTheDecimalPoint[i-1] = arrayAfterTheDecimalPoint[i-1] + 1;
+           arrayAfterTheDecimalPoint[i] = 0;
+           arrayAfterTheDecimalPoint.pop();
+       }
+   }
+
+   if(arrayAfterTheDecimalPoint[1] === 10){
+       arrayAfterTheDecimalPoint[1] = 0;
+       arrayAfterTheDecimalPoint[0] = arrayAfterTheDecimalPoint[0] + 1;
+   }
+
+   if (arrayAfterTheDecimalPoint[0] === 10) {
+       arrayBeforeTheDecimalPoint[arrayBeforeTheDecimalPoint.length - 1] = arrayBeforeTheDecimalPoint[arrayBeforeTheDecimalPoint.length - 1] + 1;
+       for (let i = numberOfPlacesBeforeTheDecimalPoint - 1; i > 0; i--) {
+           if (arrayBeforeTheDecimalPoint[i] === 10) {
+               arrayBeforeTheDecimalPoint[i - 1] = arrayBeforeTheDecimalPoint[i - 1] + 1;
+               arrayBeforeTheDecimalPoint[i] = 0;
+           }
+       }
+        arrayAfterTheDecimalPoint[0] = 0;
+       }
+   arrayAfterTheDecimalPoint[2] = ".";
+
+    let resultString  = arrayBeforeTheDecimalPoint[0].toString();
+   for (let i = 1; i < arrayBeforeTheDecimalPoint.length ; i++) {
+       resultString = resultString + arrayBeforeTheDecimalPoint[i].toString();
+   }
+   resultString = resultString + arrayAfterTheDecimalPoint[2] + arrayAfterTheDecimalPoint[0].toString() + arrayAfterTheDecimalPoint[1].toString();
+   let result = Number(resultString);
+   return(result);
```

Comment:
Wow, dit is best wel complex! Wat je had kunnen doen
```suggestion
    return number.toFixed(2);
```

Reply:
Ik deed het op deze manier, maar kreeg een onnauwkeurig resultaat. 
57.4446 zou moeten 57.45 zijn, ik kreeg 57.44.... Net als Math.round(k*100)/100) - daar kreeg ik ook 57.44. Ik weet niet, waarom. 
Dus ik zocht een andere manier:)
Regel 172